### ISSUE TITLE
In inner join with grouped sub query where condition missing 

### ISSUE THREAD

**Issue by rshkhz** [dotnet/efcore#19951]
When I join one sub query with another, where one of them is group query, then "where" condition is missing

### Steps to reproduce

Here the linq query

``` C#
  var result = await
          (from p1 in _dbContext.Persons
           join actualP in (from p in _dbContext.Persons
                                    group p by p.PersonId into g
                                    where g.Count() > 1   
                                    select new { PersonId = g.Key, Id = g.Max(x => x.Id) })
           on p1.PersonId equals actualP.PersonId
           where p1.Id != actualP.Id
           select p1.Id)
 .ToListAsync();
```

Person model is 

``` C#
internal class Person
    {
        [Column("id")]
        [Required]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [Column("person_id ")]
        [Required]
        public int PersonId { get; set; }

        [Column("updated_on")]
        [Required]
        public DateTime UpdatedOn { get; set; }
    }
```

**Translated SQL**
``` Translated SQL
SELECT t."PersonId", t."Id", p1.id
      FROM dbo.persons AS p1
      INNER JOIN (
          SELECT p.person_id AS "PersonId", MAX(p.id) AS "Id"
          FROM dbo.persons AS p
          GROUP BY p.person_id
          HAVING COUNT(*)::INT > 1
      ) AS t ON p1.person_id = t."PersonId"
--WHERE t."Id" != p1.id   --this condition will not be translated.
```
So, `WHERE` condition is missing

### Further technical details

EF Core version: 2.2.4
Database provider: Npgsql.EntityFrameworkCore.PostgreSQL
Target framework: .NET Core 2.1
Operating system:
IDE: Visual Studio 2019 16.4.4

**Comment by maumar** [comment]
This works in current bits. Producing the following sql (on sql server):

```sql
SELECT [p].[id]
FROM [Persons] AS [p]
INNER JOIN (
    SELECT [p0].[person_id ], MAX([p0].[id]) AS [c]
    FROM [Persons] AS [p0]
    GROUP BY [p0].[person_id ]
    HAVING COUNT(*) > 1
) AS [t] ON [p].[person_id ] = [t].[person_id ]
WHERE ([p].[id] <> [t].[c]) OR [t].[c] IS NULL
```

**Comment by smitpatel** [comment]
This is probably fixed in 3.1. I did not realize it was on 2.2.4. I would expect it to client eval that where in 2.x release.
